### ISSUE TITLE
Cosmetic - button on Engine Room tab

### ISSUE THREAD

**Issue by Bobbicat** [jimbob88/wheelers-wort-works#13]
You might consider this a bit nit picky but here goes:

Next to the Grist window in the Engine Room is a button that says Add New
Next to the Hop window in the Engine Room is a button that says Add Hop

Should 'Add New' read 'Add Grist', do you think?
If you think things should stay as they are I won't argue.

**Comment by jimbob88** [comment]
Add New is what is originally used by beer engine (hence it's use in my program), perhaps changing yo add grist would make it clearer?

**Comment by Bobbicat** [comment]
Hmm, you got me there. I would think it might be clearer, but sticking to Graham's original format is also very important.
Maybe, 'improvements' like that should go on a list, held in abeyance. Later a fork with these changes added could be created, so that there could be Graham's version and an alternative version, as seperate choices. 
I'd hate to be responsible for pushing this version of Beer Engine away from what it was originally intended to be.
So in my view, keep to the original format. This program works well, is great for those new to brewing and is easy to set up, without over-complication.  

As a matter of interest, Graham was not at all fond of the Tinseth and Morey calculations. Under the hood are you sticking to his methods of calculation? He reckoned his method was more accurate and reliable.

**Comment by jimbob88** [comment]
I am not using his formulas due to him never ( to the extent of my knowledge ) releasing Beer Engine (originally written in Perl I think) as Open Source. This means I do not use his formulas, some numbers I get are very close to his whereas others are miles away from his.

Perhaps if I were to be able to access his Formulas I would implement them.

**Comment by Bobbicat** [comment]
> Perhaps if I were to be able to access his Formulas I would implement them.

I have a copy of 'Home Brewing' by Graham Wheeler. On looking at the index to this book I see there are many pages devoted to 'Calculations'. His book is in copyright but I wouldn't think anyone could claim copyright for the formulae he uses and demonstrates. 
I think using formulae that Graham used when he was alive would fall under public domain, but I am no legal expert. I certainly wouldn't proceed until there was an indication that these formulae are open to being used by anyone. 
It would be awful to create an application that was declared unlawful because of the mathematics it contained.

**Comment by Bobbicat** [comment]
If you think the formulae in his book are useable, then a copy of the book is available cheaply through Amazon [£5-6 including postage if you are in the UK]. The formulae are quite extensive and well demonstrated.

**Comment by jimbob88** [comment]
I do actually own a copy of Home Brewing by Graham Wheeler, it is where I found Beer Engine originally. Do you know where the specific formula is because I didn't know it actually had it ( I believe he looks at things like colour as a linear relationship)

**Comment by Bobbicat** [comment]
I could search to make a list of these formulae Graham used, preferably from a public domain source if that is possible and meets with your approval. I am no programmer, but I can handle algebra and am willing to do a bit of internet research. However WWW is your thing and I respect that.
Personally I would like to see Grahams methods perpetuated, but that is, maybe, not what is needed here.

**Comment by jimbob88** [comment]
I feel it would be nice to have Graham's Formulae implemented as well, yet do not know where they reside perhaps if the formulae were given I could translate them into WWW

**Comment by Bobbicat** [comment]
>  ( I believe he looks at things like colour as a linear relationship)

That is what I had read too, He was quite scathing about the basis for Morey's method.
I'm not sure if I can find what Graham had to say but I will try to dig it out. I don't think Graham in any sense wanted to hide brewing secrets.

**Comment by jimbob88** [comment]
I agree 100%, somewhere I am sure these formulae are available

**Comment by Bobbicat** [comment]
On looking at the recipes in Home Brewing I see he does not mention the colour that will result from any of the recipes presented in that book. However he did outline the colour of a selection of grists. Perhaps at that stage he did not have a formula to determine that particular property.
I will pursue this, but I may be some time.

**Comment by jimbob88** [comment]
Time is not an issue in any way, if you feel at any point you do not wish to continue do not worry about it (it is just good you have helped me so much already)

**Comment by Bobbicat** [comment]
> (it is just good you have helped me so much already)

I'm so pleased someone is working with this. I am very happy to provide input.

...but, I have results!!

Graham's words in JBK Forum, outlining EBC calculation. Last post on a very short thread.

May be of use?

 https://www.jimsbeerkit.co.uk/forum/viewtopic.php?t=26000l

**Comment by jimbob88** [comment]
Yeah it could be!

**Comment by Bobbicat** [comment]
Wife says I'm like a dog with a bone.
I do like to get answers, though.

By the way Graham didn't place any restriction on the use of the information in that post, you might have noticed.

My next task - try out the latest WWW update

**Comment by jimbob88** [comment]
That link is actually pretty good, the formula I was using was pretty makeshift ( a hybrid formula ) it uses a closer to this formula now though

**Comment by Bobbicat** [comment]
Here's some more nice reference material if you have time to go through it. 
There is a further quote and explanation from Graham. It is well worth a read.

https://www.jimsbeerkit.co.uk/forum/viewtopic.php?t=63086

**Comment by jimbob88** [comment]
Thanks so much, I have been slowly reading through, it's got some very enlightening views

**Comment by Bobbicat** [comment]
I think Graham is a very reliable source for information, he doesn't, simply and blindly, copy and paste the work of others. I think his stuff is well researched.

**Comment by jimbob88** [comment]
Yeah he seems to have put a lot of work in making things his own work. Also, generally speaking, his work is well documented

**Comment by Bobbicat** [comment]
I just ran Graham's recipe for Double Diamond in WWW. 
I did his calculation manually and got a result of 19.1 EBC. 
WWW gave me 19.5. It would seem results are at least in the same area.

The results from Brewtarget are quite a bit lower, at 12.9 EBC.

**Comment by Bobbicat** [comment]
Here is another [large] acre of text from Graham, again on the subject of how beer looks.

https://www.jimsbeerkit.co.uk/forum/viewtopic.php?t=22823

**Comment by jimbob88** [comment]
It is interesting to note Series 52's lack of correlation and incorporation of true correlation with SRM/EBC

**Comment by Bobbicat** [comment]
The new and modern US has now evolved its traditionalists. Their position rests on some incorrect assumptions. To admit this is true is to undermine their position. They must and will go to any length to hold on to what they regard as infallible. This is the story of history, there is nothing new, it has happened before... etc etc.
It is a sure thing that the truth will out, but as Galileo demonstrated, sometimes you have to affirm the lie if you are to survive.
I wouldn't mind a beer engine that worked properly though

**Comment by jimbob88** [comment]
Sorry but I am not sure I understand what you mean by 

> I wouldn't mind a beer engine that worked properly though

**Comment by Bobbicat** [comment]
I have a feeling that all brewing software produces skewed results due to incorrect assumptions.
What I mean by a beer engine is a piece of software that creates a 'computer brew'. 
I would expect and hope the 'computer brew' would have similar qualities to the actual brew it represents.
I have never used Wheelers beer engine, perhaps if I had a copy and Microsoft software I would have tried it, but from what I have read of his writings I expect it would be pretty accurate.
WWW which you are engineering does seem to have this property, accuracy ,I mean.

**Comment by jimbob88** [comment]
Yeah, I have built the whole idea of WWW around Beer Engine. I myself had to run it under Wine which incorporated issues and fonts were difficult to find and gain access to. I must say I am pleased with the current product and hope to be able to iron out any bugs and release a Linux users' Beer Engine

**Comment by Bobbicat** [comment]
I think you are doing better than quite well and have every right to be pleased.
It is always great to have access to a good piece of software on Linux.

**Comment by jimbob88** [comment]
Yeah I agree, I've always loved Linux. I think the oddest bit about Beer Engine was that it was made in Linux yet never released to it...

**Comment by Bobbicat** [comment]
I'm guessing he originally made it for his own use. When he decided to share it I imagine he thought he would need to release it as a Windy exe for the general public. From what I read it gradually started to be problematic to keep on top of, from then on. As far as I understand, he reached the point that he couldnt see the point in struggling to keep it available. If he had understood open source and gone down that path I'm sure he would have had plenty of backup and support. 

Gotta go run rescue my dinner from the dog again. See ya later no doubt.

**Comment by jimbob88** [comment]
Yeah I agree. Having an Open Source project makes life a lot easier. Definitely on something like GitHub with Issue Tracking. It is easy to accumulate and serve updates to users and also be in closer contact with people. Hope your food was good once again :)
May I ask your thoughts on something? I merely wish to know what you think would be useful to be under the [wiki](https://github.com/jimbob88/wheelers-wort-works/wiki) section of WWW

**Comment by Bobbicat** [comment]
You know I'm happy to share my thoughts, I'm glad if any of it is useful.
I need some time to get some ideas together for the Wiki, but I will get back and post what I can.
I like the new html save formatting, btw.

**Comment by Bobbicat** [comment]
I've created a new Issue to cover what might be included in a useful wiki. I'll add to it as I get content together.

On a slightly different tack, if you put the components of a recipe from Brew Your Own British Real Ale by Graham into WWW and it works out exactly as he outlines it then I think you will have pretty much cloned the calculator in Beer Engine. He will have used that software to set up his recipes, I am sure.

**Comment by jimbob88** [comment]
I have just opened the revised edition of "Brew your own British Real Ale" and he states that the original gravity is that quoted from the brewery and if that wasn't available it has been reverse engineered from the alcohol. Anyway here are the results:
This is my setup:
![Screenshot from 2019-03-31 10-23-59](https://user-images.githubusercontent.com/9913366/55287308-1e041400-539f-11e9-8d15-63e788c9e423.png)
### Banks' Hanson's Mild
| Fermentable Ingredient | 23 litres |   | Hop                  | 23 litres |
|------------------------|-----------|---|----------------------|-----------|
| Pale Malt (grams)      | 3190      |   | Fuggle Hops (grams)  | 21        |
| Crystal Malt (grams)   | 360       |   | Golding Hops (grams) | 18        |
| Black Malt (grams)     | 69        |   |                      |           |

Graham's results are as follows:

| Original Gravity | Final Gravity | ABV  | IBU | Colour |
|------------------|---------------|------|-----|--------|
| 1035             | 1009          | 3.4% | 25  | 50     |

Whereas Wheeler's Wort Works creates:

| Original Gravity | Final Gravity | ABV  | IBU | Colour |
|------------------|---------------|------|-----|--------|
| 1034.5           | 1009.2        | 3.3% | 24  | 49.7   |

Graham's Beer Engine creates:

| Original Gravity | Final Gravity | ABV  | IBU | Colour |
|------------------|---------------|------|-----|--------|
| 1034             | 1009          | 3.3% | 25  | 50     |


**Interesting Note:** if you open Graham's Beer Engine's save file in Wheeler's Wort Works you get 25 IBUs as expected by the prediction.

**Comment by Bobbicat** [comment]
So I don't think anyone could argue with those results. I suppose the next step would be try an actual brew and compare with that. From what I have read Graham disputed IBU  calculations, but as he hadn't a solid formula to offer as a substitue he deferred from saying anything. He had said the issue had been enough to stop him updating Beer Engine and authoring another book.

My set up isn't exactly standard but it does work for me. I first mill my grist. I get an efficiency between 80 and 90%. My method comes from information I have come across on line. It is very 'trial and error', what works I keep. I don't pretend that what works for me is good for everyone.
I tend to make small one gallon brews. I mash for only 30 minutes at 66-67C in a bag then dunk the bag in 'sparge' water for 15 minutes. I combine the mash and sparge and boil for 30 minutes. I use a hop bag in the boil. I use a coil to cool, then remove the hop bag and aerate by stirring vigorously. I pitch the yeast directly into the cooled 'copper', cover and let the fermentation proceed.
Once fermentation is complete I syphon into a keg, although until recently I was bottling.

I have only been all grain brewing for a short time and I wouldn't call myself an expert by any means.

Where's this taking us? Well, I keep notes on my brews and have 'tried them out' in WWW. The application follows closely the actual results I have experienced and noticed. 
By the way I have also found that my results and those produced by other software do not tally. Hence my search for something that would perform and predict better.

I am using Fawcett's grist [it what my local home brew shop has in stock] and on the maltster's site is an extensive database with enough information to tweak up WWW's settings. I've also edited my hops. I suppose some variation will occur if you don't use accurate data.

**Comment by jimbob88** [comment]
Good to hear WWW has been working well for you. I have always found it odd with Brewing Software how they all seem to want to change formulas and they are all constantly changed yet never seem to have all the data values conforming... Also Brewing Software often is uneasy/unwieldy at first use, with many hidden menus which is something I've always respected about Beer Engine - everything was in front of you without complication
